python/ray/node.py

```diff
@@ -803,9 +820,12 @@ def start_reaper_process(self):
 
     def start_redis(self):
         """Start the Redis server."""
+        external_redis = self._ray_params.external_addresses
+        if use_gcs_for_bootstrap() and external_redis is None:
+            return
```

Comment:
@mwtian  we'll not start redis even external-address is given. the redis service should be started by the user.

we'll not deliver redis with ray pkg in long term.

Reply:
The name of this function is not the most precise. With external Redis configured, `start_redis()` only configures Ray to use the external Redis, and does not actually start Redis. It will be renamed after migration is complete.

Reply:
I see. The naming is confusing and also start_redis code looks bad to mix setup and start in the same place.

could you add comments in the code about that it wont' start redis process?

btw, do you know what's the purpose of RedisShard there?

Reply:
The number and details of Redis shards are used in the Redis retry logic. I tried to avoid changing anything about them.